integration/tests/neon_evm/solana_native/test_scheduled_trx_balance.py

```diff
@@ -0,0 +1,499 @@
+        neon_user.neon_address.hex(), nonce, evm_loader.sol_chain_id
+    ).balance
+
+    holder_acc_balance_tree_destroyed = evm_loader.get_solana_balance(holder_acc)
+    operator_balance_tree_destroyed = evm_loader.get_operator_neon_balance(operator_keypair, evm_loader.sol_chain_id)
+    neon_user_balance_tree_destroyed = evm_loader.get_solana_balance(neon_user.solana_account.pubkey())
+    treasury_pool_balance_tree_destroyed = evm_loader.get_solana_balance(treasury_pool.account)
+    tree_acc_balance_tree_destroyed = evm_loader.get_solana_balance(tree_acc)
+
+    assert tree_acc_balance_tree_destroyed == 0, "Tree_acc balance's supposed to be 0"
+    treasury_pool_final_balance_no_fees = treasury_pool_balance_tree_destroyed - deposit_to_tree_acc * trx_count
+    assert (
+        treasury_pool_balance == treasury_pool_final_balance_no_fees
+    ), f"Expected {treasury_pool_balance}, but got {treasury_pool_final_balance_no_fees}"
+    assert (
+        tree_acc_balance_inner_tree_destroyed == 0
+    ), "Tree acc inner balance after destroyed failed, expected to be zero"
+
+    print(f"Delta neon_user {neon_user_balance_tree_destroyed}")
+    print(f"Delta operator {operator_balance_tree_destroyed}")
+    print(f"Delta holder {holder_acc_balance_tree_destroyed}")
+    print(f"Delta treasury {treasury_pool_balance_tree_destroyed}")
+    print(f"Tree Acc balance {tree_acc_balance_tree_destroyed}")
+
+
+def test_failed_trx_with_outer_deposit(
+    neon_user, neon_api_client, evm_loader, operator_keypair, treasury_pool, basic_contract, solana_account
+):
+
+    # trx_status: failed
+    # user_balance: outer deposit non zero
+    # tree_acc: two scheduled trx in tree acc
+
+    evm_loader.create_balance_account(neon_user.checksum_address, neon_user.solana_account, evm_loader.sol_chain_id)
+
+    neon_user_balance_sol = evm_loader.get_neon_balance(neon_user.neon_address, evm_loader.sol_chain_id)
+    assert neon_user_balance_sol == 0, f"Inner sol balance is {neon_user_balance_sol}, but has to be zero"
+
+    holder_acc = evm_loader.create_holder(operator_keypair)
+
+    holder_acc_balance = evm_loader.get_solana_balance(holder_acc)
+    operator_balance = evm_loader.get_operator_neon_balance(operator_keypair, evm_loader.sol_chain_id)
+    neon_user_balance_before = evm_loader.get_solana_balance(neon_user.solana_account.pubkey())
+    treasury_pool_balance = evm_loader.get_solana_balance(treasury_pool.account)
+
+    print(f"{neon_user_balance_before=}")
+    print(f"{neon_user_balance_sol=}")
+    print(f"{operator_balance=}")
+    print(f"{treasury_pool_balance=}")
+    print(f"{holder_acc_balance=}")
+
+    nonce = evm_loader.get_neon_nonce(neon_user.neon_address, evm_loader.sol_chain_id)
+    tx0 = ScheduledTransaction(
+        neon_user.neon_address,
+        None,
+        nonce,
+        index=0,
+        target=basic_contract.eth_address,
+        value=0,
+        call_data=b"",
```

Comment:
let's make here some real transaction that fails during execution (for example assert inside solidity code)